Classes/Hooks/HookService.php

```diff
@@ -0,0 +1,114 @@
+<?php
+declare(strict_types=1);
+
+namespace OliverKlee\Seminars\Hooks;
+
+use UnexpectedValueException;
+
+use OliverKlee\Seminars\Interfaces\Hook;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * This class provides functions for unified hooks.
+ *
+ * It unifies the wide-spread use of this type of hooks in seminars.
```

Comment:
This description still reads kind of anemic to me. I'd prefer something about what the class does.